testsuite/classlibrary/TestDocument.sc

```diff
@@ -0,0 +1,55 @@
+TestDocument : UnitTest {
+	// autorun doesn't have access to this object's scope
+	// need a public place to communicate the result
+	classvar <>success;
+
+	test_new_document_runs_initAction {
+		var success = false, doc, save;
+		if(Platform.ideName == "scqt") {
+			save = Document.initAction;
+			protect {
+				Document.initAction = { success = true };
+				doc = Document.new;
+				// failureMessage is deliberately nil
+				// we will assert below, we don't need `wait` to call `failed` itself
+				this.wait({ success }, failureMessage: nil, maxTime: 0.2);
+				doc.close;
+			} {
+				Document.initAction = save;
+			};
+			this.assert(success, "Document.new should fire Document.initAction function");
+		} {
+			// TODO: skip
+		};
+	}
+
+	test_open_autorun_document_runs_code {
+		var path = Platform.defaultTempDir +/+ "autoRunTest%.scd".format(UniqueID.next),
```

Comment:
what's the intent here with `UniqueID.next`?

Reply:
Hm, I think my intent was to reduce the likelihood of crashing into a leftover file in the temp directory. But UniqueID probably wouldn't work for that.

I'll just take it out.